From 8b91e44cdcc73e4534e052e9de7818c6d89315d8 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Thu, 23 Feb 2012 07:54:36 +0100
Subject: [PATCH mediastreamer2] Do not destroy uninitialized echo canceller

Under certain circumstances it can happen that the echo canceller is not
initialized before it is destroyed. The ms_filter_destroy() function
does not check for a NULL pointer before dereferencing so we must make
sure that the caller does not pass one in.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 src/audiostream.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/mediastreamer2/src/audiostream.c b/mediastreamer2/src/audiostream.c
index 2274f0d..9ac83d0 100644
--- a/mediastreamer2/src/audiostream.c
+++ b/mediastreamer2/src/audiostream.c
@@ -389,7 +389,7 @@ int audio_stream_start_full(AudioStream *stream, RtpProfile *profile, const char
 	ms_filter_call_method(stream->soundwrite,MS_FILTER_SET_NCHANNELS, &tmp);
 
 	/*configure the echo canceller if required */
-	if (!use_ec) {
+	if (!use_ec && stream->ec != NULL) {
 		ms_filter_destroy(stream->ec);
 		stream->ec=NULL;
 	}
-- 
1.7.9.1

